src/neo-vm/Utility.cs

```diff
@@ -25,6 +25,26 @@ static Utility()
             StrictUTF8.EncoderFallback = EncoderFallback.ExceptionFallback;
         }
 
+        public static BigInteger ModInverse(this BigInteger n, BigInteger modulus)
+        {
+            if (n <= 0) throw new ArgumentOutOfRangeException(nameof(n));
+            if (modulus < 2) throw new ArgumentOutOfRangeException(nameof(modulus));
+            BigInteger m = modulus, v = 0, d = 1;
+            while (n > 0)
+            {
+                BigInteger t = m / n, x = n;
+                n = m % x;
+                m = x;
+                x = d;
+                d = checked(v - t * x);
+                v = x;
+            }
+            v %= modulus;
+            if (v < 0) v += modulus;
+            if (!(n * v % modulus).IsOne) throw new InvalidOperationException();
```

Comment:
```suggestion
            if (!(num * v % modulus).IsOne) throw new InvalidOperationException();
```

@shargon This should fix the problem, I guess... Original value `n` should be kept for usage here on the checks... note that `v` seem to store the result, but `n` is destroyed during the processing (becoming zero), and that's why test should not be working. I could not test here on my computer, but it should work.